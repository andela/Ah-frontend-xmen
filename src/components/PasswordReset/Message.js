import React from 'react';


const message = props => (

  <div className="container text-center bg-secondary">
    <div className="card-body px-lg-5 py-lg-5">
      <h1>Password Reset</h1>
      <div className="text-center mb-5">
        {props.message}
      </div>

      <div className="col-md-4 text-center">
        <div className="text-center text-primary">
          {props.title}
        </div>
        <div className="tect-muted" />
      </div>

    </div>
  </div>


);

export default message;
