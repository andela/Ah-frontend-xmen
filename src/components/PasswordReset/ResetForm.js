import React from 'react';

const ResetForm = props => (

  <div className="container text-center bg-secondary">
    <div className="card-body px-lg-5 py-lg-5">
      <h1>Password Reset</h1>
      <div className="text-center mb-5">
        Enter your Authors haven email address that you used to register.
        {' '}
        We will send you an email with a link to reset your password.
      </div>
      <form onSubmit={props.onSubmit}>
        <div className="form-group mb-3">
          <div className="input-group input-group-alternative">
            <div className="input-group-prepend">
              <span className="input-group-text"><i className="ni ni-email-83" /></span>
            </div>
            <input className="form-control" placeholder="Email" type="email" required />
          </div>
        </div>

        <div className="text-center">
          <button type="submit" className="btn button-primary my-4">Request Link</button>
        </div>

      </form>
    </div>
  </div>


);

export default ResetForm;
