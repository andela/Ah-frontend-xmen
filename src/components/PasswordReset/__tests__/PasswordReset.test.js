import React from 'react';
import { shallow } from 'enzyme';
import ResetForm from '../ResetForm';
import NewPasswordForm from '../NewPasswordFrom';
import Message from '../Message';

describe('test password reset components', () => {
  it('should render ResetForm without crashing', () => {
    const wrapper = shallow(<ResetForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render NewPasswordForm without crashing', () => {
    const wrapper = shallow(<NewPasswordForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Message without crashing', () => {
    const wrapper = shallow(<Message />);
    expect(wrapper).toMatchSnapshot();
  });
});
