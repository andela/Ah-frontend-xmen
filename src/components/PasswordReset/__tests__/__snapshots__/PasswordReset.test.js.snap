// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test password reset components should render Message without crashing 1`] = `ShallowWrapper {}`;

exports[`test password reset components should render NewPasswordForm without crashing 1`] = `ShallowWrapper {}`;

exports[`test password reset components should render ResetForm without crashing 1`] = `ShallowWrapper {}`;
